File: tournaments/minimalNumberOfCoins/minimalNumberOfCoins.py
def minimalNumberOfCoins(coins, price):
    i = -1
    numCoins = 0
    
    while 0 < price and len(coins) >= abs(i):
        if price >= coins[i]:
            price -= coins[i]
            numCoins += 1
        else:
            i -= 1
    
    return numCoins

File: tournaments/minimalNumberOfCoins/minimalNumberOfCoins.cpp
int minimalNumberOfCoins(std::vector<int> coins, int price) {

  int result = 0;

  for (int i = (int)coins.size() - 1; i >= 0; i--) {
    result += price / coins[i];
    price %= coins[i];
  }

  if (price) {
    return -1;
  }
  return result;
}
